run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
+	seen := map[string]string{}
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// Checks if outputVariables are defined in terms of other outputVariables.
+// Creates a slice of unique model variables that are required to calculate
+// the requested outputVariables.
+func checkForDerivitives(m map[string]string) (map[string]string, []string) {
+	getVariables := []string{}
+	for key, val := range m {
+		expression, _ := govaluate.NewEvaluableExpression(val)
+		uniqueVars := removeDuplicates(expression.Vars())
```

Comment:
I think this change is worth at least checking to see if the main package is willing to add it. I made a couple of suggestions for things to change in your version. It would probably also be good to add a test, which you could model after [this one](https://github.com/Knetic/govaluate/blob/master/parsing_test.go#L1360). Then you can just create a pull request and see what they say.